internal/ingress/controller/nginx.go

```diff
@@ -262,10 +263,39 @@ func (n *NGINXController) Start() {
 
 	n.store.Run(n.stopCh)
 
-	if n.syncStatus != nil {
-		go n.syncStatus.Run()
+	// we need to use the defined ingress class to allow multiple leaders
+	// in order to update information about ingress status
+	electionID := fmt.Sprintf("%v-%v", n.cfg.ElectionID, class.DefaultClass)
+	if class.IngressClass != "" {
+		electionID = fmt.Sprintf("%v-%v", n.cfg.ElectionID, class.IngressClass)
 	}
 
+	setupLeaderElection(&leaderElectionConfig{
+		Client:     n.cfg.Client,
+		ElectionID: electionID,
+		OnStartedLeading: func(stopCh chan struct{}) {
+			if n.syncStatus != nil {
+				go n.syncStatus.Run(stopCh)
+			}
+
+			n.metricCollector.OnStartedLeading(electionID)
+		},
+		OnStoppedLeading: func() {
+			n.metricCollector.OnStoppedLeading(electionID)
+
+			// Remove prometheus metrics related to SSL certificates
```

Comment:
Why is this not part of `metricCollector.OnStoppedLeading`? This seems like something `metricCollector` should be concerned

Reply:
done